src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hosting/ConfigurableLanguageServerFeatureOptions.cs

```diff
@@ -39,6 +40,7 @@ internal class ConfigurableLanguageServerFeatureOptions : LanguageServerFeatureO
     public override bool DisableRazorLanguageServer => _disableRazorLanguageServer ?? _defaults.DisableRazorLanguageServer;
     public override bool ForceRuntimeCodeGeneration => _forceRuntimeCodeGeneration ?? _defaults.ForceRuntimeCodeGeneration;
     public override bool UseNewFormattingEngine => _useNewFormattingEngine ?? _defaults.UseNewFormattingEngine;
+    public override bool AvoidExplicitCommitCharacters => _avoidExplicitCommitCharacters ?? _defaults.AvoidExplicitCommitCharacters;
```

Comment:
Is this going to be user configurable? If not we could just hard code to true for VS Code and false for VS

Reply:
Good question, my current thinking is "no" to user-configurable. However theoretically a different client could invoke rzls, right?

Reply:
Yes but it would have to be spec specific. I don't think soft selection is in the LSP spec right? Otherwise we'd just check client capabilities. 

My thinking is really: wiring this up is very VS specific. If we have a request for a client to add soft selection and set this we could reconsider the design and possibly check a specific client capability rather than passing into the executable itself.

In fact a purist fix would say that this should be added to the `VSInternalClientCapabilities` and check that way. I won't push for that because I know it's a lot of overhead. That said I do think we should try to limit these options to purely "configuration" oriented things and not client capability checks.

Reply:
Bonus is there's no follow up on insertion in VS Code ;)

Reply:
That's good way of looking at it. I went ahead and followed your suggestion here, and not needing to make VSCode changes is certainly a nice bonus :)